src/main/java/mil/dds/anet/resources/ReportResource.java

```diff
@@ -747,12 +774,57 @@ public Response showRollupEmail(@Auth Person user, @QueryParam("startDate") Long
 			return Utils.resultGrouper(list, "stats", "personId", tlf);
 		}
 	}
-
+	
 	private Map<Integer, Organization> getOrgsByShortNames(List<String> orgShortNames) {
-		final Map<Integer, Organization> result = new HashMap<>();
-		for (final Organization organization : engine.getOrganizationDao().getOrgsByShortNames(orgShortNames)) {
-			result.put(organization.getId(), organization);
+			final Map<Integer, Organization> result = new HashMap<>();
+			for (final Organization organization : engine.getOrganizationDao().getOrgsByShortNames(orgShortNames)) {
+				result.put(organization.getId(), organization);
+			}
+			return result;
+		}
+	/**
+	 * The comparator to be used when ordering the roll up graph results to ensure
+	 * that any pinned organisation names are returned at the start of the list.
+	 * 
+	 * @author timothy.ward
+	 *
+	 */
+	public static class RollupGraphComparator implements Comparator<RollupGraph> {
+
+		private final List<String> pinnedOrgNames;
+
+		/**
+		 * Creates an instance of this comparator using the supplied pinned organisation
+		 * names.
+		 * 
+		 * @param pinnedOrgNames
+		 *            the pinned organisation names
+		 */
+		public RollupGraphComparator(final List<String> pinnedOrgNames) {
+			this.pinnedOrgNames = pinnedOrgNames;
+		}
+
+		/*
+		 * (non-Javadoc)
+		 * 
+		 * @see java.util.Comparator#compare(java.lang.Object, java.lang.Object)
+		 */
+		@Override
+		public int compare(final RollupGraph o1, final RollupGraph o2) {
+
+			if (pinnedOrgNames.contains(o1.getOrg().getShortName())) {
```

Comment:
Probably nicer to extract these two:
```
final Organization org1 = o1.getOrg();
final Organization org2 = o2.getOrg();
```
and use them further down.

Reply:
If the method calls were longer such as a.getOrg().getName().getXXX().getYYY() I would agree, but in this case I do not see any benefit in the extraction.